tools/cdb2sql/cdb2sql.cpp

```diff
@@ -943,31 +966,52 @@ static void print_column(FILE *f, cdb2_hndl_tp *hndl, int col)
         return;
     }
 
-    switch (cdb2_column_type(hndl, col)) {
+    int type = cdb2_column_type(hndl, col);
+    std::string typeStr;
+
+    if (printmode & DISP_COLTYPE) {
+        if (type >= MIN_COL_TYPE_NAME && type <= MAX_COL_TYPE_NAME) {
+            typeStr = col_type_names[type];
```

Comment:
`col_type_names[type] + " "` should do as well.

Reply:
Just tried that, it does not appear to work (it won't compile).